net/nimble/host/mesh/src/health.c

```diff
@@ -319,6 +319,7 @@ static void health_period_set_unrel(struct bt_mesh_model *model,
 	BT_DBG("period %u", period);
 
 	srv->period = period;
+	model->pub->period_div = srv->period;
```

Comment:
I went with a different approach for this: the health server context period divider tracking (srv->period) is in practice redundant since the same information is already stored in the generic model publication context (model->pub->period_div). Removing the health-model specific value and just using the generic one solves the problem equally well. I'll provide a reference to my Zephyr patch here as soon as I upload it.